bluesky/tests/test_run_engine.py

```diff
@@ -571,34 +571,41 @@ def simple_plan(motor):
     assert motor.position == 1024
 
 
+@pytest.mark.skipif(
+    os.environ.get('TRAVIS', None) == 'true' and sys.platform == 'darwin',
+    reason=("The file-descriptor wake up based signal handling "
+            "does not work on travis on OSX"))
 def test_sigint_three_hits(RE, hw):
     import time
     motor = hw.motor
-    motor.delay = 1
+    motor.delay = .5
 
     pid = os.getpid()
 
-    def sim_kill(n=1):
+    def sim_kill(n):
         for j in range(n):
-            time.sleep(.02)
+            time.sleep(.05)
             os.kill(pid, signal.SIGINT)
 
     lp = RE.loop
     motor.loop = lp
-    start = time.monotonic()
-    threading.Timer(.05, sim_kill, (3,)).start()
-    threading.Timer(.1, sim_kill, (3,)).start()
-    threading.Timer(.2, sim_kill, (3,)).start()
+
+    def self_sig_int_plan():
+        threading.Timer(.05, sim_kill, (3,)).start()
```

Comment:
:+1: Makes more sense to do it like this - start the timer only after the plan starts. Could also do it with a `threading.Event` or so.

Reply:
Yeah, all of the sigint tests could be cleaned up.